tarantool/connection.py

```diff
@@ -298,6 +329,52 @@ def connect_unix(self):
             self.connected = False
             raise NetworkError(e)
 
+    def wrap_socket_ssl(self):
+        '''
+        Wrap an existing socket with SSL socket.
+
+        :raise: NetworkError
+        :raise: `ssl.SSLError`
+        '''
+        if not is_ssl_supported:
+            raise NetworkError("SSL is unsupported by the python.")
+
+        ver = sys.version_info
+        if ver[0] < 3 or (ver[0] == 3 and ver[1] < 5):
+            raise NetworkError("SSL transport is supported only since " +
+                               "python 3.5")
+
+        if hasattr(ssl, 'TLSVersion'):
+            # Since python 3.7
+            context = ssl.SSLContext(ssl.PROTOCOL_TLS)
+            # Require TLSv1.2, because other protocol versions don't seem to
+            # support the GOST cipher.
+            context.minimum_version = ssl.TLSVersion.TLSv1_2
+            context.maximum_version = ssl.TLSVersion.TLSv1_2
+        else:
+            # Deprecated, but it works for python < 3.7
+            context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
+
+        if self.ssl_cert_file:
+            def password_raise_error():
+                raise NetworkError("a password for decrypting the private" +
+                                   "key is unsupported")
+            context.load_cert_chain(certfile=self.ssl_cert_file,
+                                    keyfile=self.ssl_key_file,
+                                    password=password_raise_error)
```

Comment:
Can you clarify this line?

Reply:
[The reason is described in @Totktonada comment.](https://github.com/tarantool/tarantool-python/pull/220#discussion_r898462555)

Reply:
I agree that it would be good to have a brief explanation in the code.

Reply:
I've added a comment:
```patch
--- a/tarantool/connection.py
+++ b/tarantool/connection.py
@@ -366,6 +366,14 @@ class Connection(ConnectionInterface):
                 context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
 
             if self.ssl_cert_file:
+                # If the password argument is not specified and a password is
+                # required, OpenSSL’s built-in password prompting mechanism
+                # will be used to interactively prompt the user for a password.
+                #
+                # We should disable this behaviour, because a python
+                # application that uses the connector unlikely assumes
+                # interaction with a human + a Tarantool implementation does
+                # not support this at least for now.
```